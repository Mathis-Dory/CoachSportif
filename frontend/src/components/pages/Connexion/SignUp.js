import React from 'react';
import '../../../App.css';
import FormSignup from "../../FormSignUp"

export default function SignUp() {
    return (
        <>
            <FormSignup/>;
        </>

    );
}
