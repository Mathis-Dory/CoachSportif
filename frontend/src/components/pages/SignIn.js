import React from 'react';
import '../../App.css';
import FormSignin from "../FormSignIn"

export default function SignIn() {
    return (
        <>
            <FormSignin/>;
        </>

    );
}