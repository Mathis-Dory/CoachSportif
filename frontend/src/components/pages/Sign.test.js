import ReactDOM from "react-dom";
import {MemoryRouter} from "react-router-dom";
import SignUp from "./SignUp";
import SignIn from "./SignIn";
import React from "react";
import TestRenderer from "react-test-renderer";

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><SignUp /></MemoryRouter>, div);
});

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><SignIn /></MemoryRouter>, div);
});

it("renders a form", () => {
  const testRenderer = TestRenderer.create(<MemoryRouter><SignUp /></MemoryRouter>);
  const testInstance = testRenderer.toJSON();
  expect(testInstance).toMatchSnapshot();
});


