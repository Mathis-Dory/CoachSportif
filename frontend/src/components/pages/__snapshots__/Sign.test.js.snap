// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form 1`] = `
Array [
  <div
    className="form-content"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          ALIAS
        </label>
        <input
          className="form-input"
          name="alias"
          onChange={[Function]}
          placeholder="Entrez votre alias"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          EMAIL
        </label>
        <input
          className="form-input"
          name="email"
          onChange={[Function]}
          placeholder="Entrez votre e-mail"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          NAISSANCE
        </label>
        <input
          className="form-input"
          name="naissance"
          onChange={[Function]}
          required={true}
          type="date"
          value=""
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          TAILLE
        </label>
        <input
          className="form-input"
          min="100"
          name="alias"
          onChange={[Function]}
          placeholder="Entrez votre taille"
          required={true}
          type="number"
          value={160}
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          POIDS
        </label>
        <input
          className="form-input"
          min="30"
          name="alias"
          onChange={[Function]}
          placeholder="Entrez votre poids"
          required={true}
          type="number"
          value={40}
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          MOT DE PASSE
        </label>
        <input
          className="form-input"
          name="password"
          onChange={[Function]}
          pattern=".{6,}"
          placeholder="Entrez mot de passe de plus de 6 caractères"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        class="form-check form-switch"
      >
        <input
          id="agree"
          onChange={[Function]}
          required={true}
          type="checkbox"
        />
        <label
          htmlFor="agree"
        >
          J'ACCEPTE 
          <b>
            <a
              href="/conditions"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                }
              }
            >
               LES CONDITIONS GÉNÉRALES ET LA POLITIQUE DE CONFIDENTIALITÉ 
            </a>
          </b>
        </label>
      </div>
      <button
        className="form-input-btn"
        type="submit"
      >
        S'INSCRIRE
      </button>
    </form>
  </div>,
  ";",
]
`;
